### PR TITLE
Add UI job position

### PR DESCRIPTION
**Pull request by WalterGoldenOwl**
Description Pull Request:
## Todo
- Code UI add Job Position

## Task
Link of task: https://trello.com/c/pRw3dBe1

## Screenshot
Screenshot_1696842820

### CODE REVIEW

File: lib/src/presentations/job_position/cubit/job_position_cubit.dart [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+part 'job_position_state.dart';
+
+@injectable
+class JobPositionCubit extends Cubit<JobPositionState> {
+  final SearchJobPositionUseCase _searchJobPositionUseCase;
+  TextEditingController searchController = TextEditingController();
+
+  JobPositionCubit(this._searchJobPositionUseCase)
+      : super(const JobPositionState(text: "")) {
```
Comment (dustin-goldenowl) [inline lib/src/presentations/job_position/cubit/job_position_cubit.dart]: Should be use JobPositionState.ds()

File: lib/src/presentations/job_position/cubit/job_position_state.dart [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  final DataState<List<String>>? dataState;
+  final String text;
+
+  JobPositionState copyWith({
+    DataState<List<String>>? dataState,
+    String? text,
+  }) {
+    return JobPositionState(dataState: dataState, text: text ?? this.text);
+  }
+
```
Comment (dustin-goldenowl) [inline lib/src/presentations/job_position/cubit/job_position_state.dart:15]: Should be define factory class for set default value in cubit.
Example: 
```
factory JobPositionState.ds(){
return JobPositionState(text: "");
}
```

File: lib/src/presentations/job_position/cubit/job_position_cubit.dart (hunk 1 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/job_position/cubit/job_position_cubit.dart]: I fixed it

File: lib/src/presentations/job_position/cubit/job_position_state.dart (hunk 2 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/job_position/cubit/job_position_state.dart:15]: I fixed it

**Pull request merged**